app/src/main/java/bg/zahov/app/ui/workout/WorkoutScreen.kt

```diff
@@ -0,0 +1,768 @@
+                modifier = Modifier.padding(start = 16.dp),
+                style = MaterialTheme.typography.titleLarge,
+                color = MaterialTheme.colorScheme.onSecondaryContainer
+            )
+
+            WorkoutScreenInputField(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(12.dp),
+                value = note,
+                onValueChanged = { onNoteChange(it) },
+                label = {
+                    Text(
+                        text = stringResource(R.string.add_note),
+                        style = MaterialTheme.typography.labelMedium,
+                        color = MaterialTheme.colorScheme.secondary
+                    )
+                }
+            )
+            /**
+             * This structure ensures that the buttons placed within the Column that surrounds
+             * the LazyColumn are always visible
+             */
+            Column(modifier = Modifier.fillMaxWidth()) {
+                LazyColumn(
+                    modifier = Modifier
+                        .weight(1f)
+                        .padding(start = 8.dp, end = 8.dp, bottom = 8.dp),
+                    verticalArrangement = Arrangement.spacedBy(4.dp),
+                    horizontalAlignment = Alignment.CenterHorizontally
+                ) {
+                    itemsIndexed(
+                        items = exercises,
+                        key = { index, item -> if (item is WorkoutEntry.ExerciseEntry) item.id else (item as WorkoutEntry.SetEntry).id }) { index, it ->
+
+                        when (it) {
+                            is WorkoutEntry.ExerciseEntry -> {
+                                Exercise(
+                                    modifier = Modifier.animateItem(fadeOutSpec = spring(stiffness = Spring.StiffnessLow)),
+                                    it.name,
+                                    note = it.note,
+                                    floatArrangement = weightValues,
+                                    labelColor = MaterialTheme.colorScheme.onSecondaryContainer,
+                                    onExerciseNoteChange = { onExerciseNoteChange(index, it) },
+                                    onAddSet = { onAddSet(index) },
+                                    onReplaceExercise = { onReplaceExercise(index) },
+                                    onRemoveExercise = { onRemoveExercise(index) }
+                                )
+                            }
+                            /**
+                             * Replicating a debounce effect because otherwise the
+                             * after the threshold of the swipe has been reached
+                             * the set just clips out and it made it look very choppy
+                             */
+                            is WorkoutEntry.SetEntry -> {
+                                val scope = rememberCoroutineScope()
```

Comment:
don't define variables which will be reinitialized on every recomposition lift them up

Reply:
👍